### ISSUE TITLE
How to inject current locale into the store?

### ISSUE THREAD

**Issue by subzero911** [aissat/easy_localization#304]
I wrap my multi-provider into EasyLocalization widget and try to inject current locale into the store constructor (I'm using MobX stores)

![image](https://user-images.githubusercontent.com/12999702/103211018-298bc280-4918-11eb-92b9-7a255e5d0719.png)

This leads to the following error:

![image](https://user-images.githubusercontent.com/12999702/103211057-39a3a200-4918-11eb-9a30-a11d4daca0e2.png)

That makes sense because the locale can change during runtime. But how can I get access to the current locale in the store (without context)? I need the locale because I perform HTTP request with "locale" parameter in the store.

**Comment by subzero911** [comment]
I spend the whole lot of time and came up with a workaround.
I divide my widget into two, and read a locale in didChangeDependencies() and put it into providers' constructors.
![image](https://user-images.githubusercontent.com/12999702/103276886-bbf39b00-49d8-11eb-86ad-1345de1a0f5e.png)

**Comment by Overman775** [comment]
@subzero911 yep, it needs a split context. It is not for nothing that it is divided into examples. You can use `didChangeDepencies` like upper or `Builder` widget

....
assetLoader:  YamlLoader()
